src/app/lib/device/airplay.js

```diff
@@ -1,56 +1,124 @@
 (function (App) {
     'use strict';
 
-    var airplayer = require('airplayer'),
-        netw = require('network-address'),
-        list = airplayer(),
+    /*
+      Useful AirPlay protocol references:
+      - "Unofficial AirPlay Protocol Specification" (https://nto.github.io/AirPlay.html)
+      - @watson's "airplay-protocol" (https://github.com/watson/airplay-protocol)
+    */
+    var Bonjour = require('bonjour'),
+        AirPlayProtocol = require('airplay-protocol');
+
+    var bonjourBrowser = new Bonjour(),
         collection = App.Device.Collection;
 
-    var makeID = function (baseID) {
-        return 'airplay-' + baseID.replace('.', '');
+    var makeID = function (bonjourService) {
+        // strong de-duplication ID since there can only be 1 bonjour service per host+port
+        var uniqueDeviceId = bonjourService.host + '_' + bonjourService.port;
+        return 'airplay-' + Common.md5(uniqueDeviceId);
```

Comment:
@xaiki Just explain why Chromecast uses md5?

https://github.com/butterproject/butter-desktop/blob/master/src/app/lib/device/chromecast.js#L14

We do not need the internal de-duplication ID to be human readable.

Imagine this: We have a device that is bugged. We want to debug it. Do we look at device.id? No. We look at device.service.name and device.service.host and device.service port.

I think we should keep Common.md5(). It's a nicer variable name. And all debug info is available to us anyway.